### PR TITLE
Parametrise backend using a config json file

### PR DESCRIPTION
**Pull request by VFjr**
# Summary

This solves issue #82

Main now loads the runtime parameters from a json file, and puts them inside a `ConfigParameters` struct that contains pretty much all parameters needed. 

New `config` package created for the `ConfigParameters` struct and the `LoadParamFromJson` function

The `LoadParamFromJson` function  is used to parse the json and convert it into a struct and is also hopefully very similar to what will be required for the frontend communication to parse the json ( issue #81 )

## Additional Information

Also added functionality for command-line flags to allow the used to choose the location of the config file, however if none is provided, then it will use the config.json file which already contains the default values of previous commit.

### CODE REVIEW

File: cmd/backend/main.go
Comment (udai-arneja) [inline cmd/backend/main.go:54]: I would argue it's worth passing in a pointer to `parameters` to both `simulation.NewSimEnv` and `health.NewHealthInfo` and setting the parameters there. Makes it more readable imo.

File: config.json
Comment (udai-arneja) [inline config.json:3]: maybe making this a little more interpretable, like `"randomAgent": int` and then in the `configparameters.go`, `func LoadParamFromJson` put it into the list in the right places? Makes it a little easier for people to input their agents etc/more readable.

**Review by udai-arneja — CHANGES_REQUESTED** [review]
Looks good overall, thanks for this. Few minor comments.

**Comment by MattScottEEE** [comment]
More of a general comment, but do the command line arguments come with a "-h" parameter, giving a list of the possible arguments? I'm not sure if the flag package handles this implicitly

**Comment by VFjr** [comment]
> More of a general comment, but do the command line arguments come with a "-h" parameter, giving a list of the possible arguments? I'm not sure if the flag package handles this implicitly

Yes, it does that automatically. Also thanks for mentioning it, while testing this I spotted an error in my flag implementation.
image

**Review by MattScottEEE — COMMENTED** [review]
Overall LGTM, but will let infra team give explicit approval if I've missed something

**Comment by VFjr** [comment]
Oof, tried to rebase with github desktop and looks like I messed something up

**Comment by JaafarRammal** [comment]
> Oof, tried to rebase with github desktop and looks like I messed something up
Why is GitHub desktop a thing, we should all use VIM

**Comment by VFjr** [comment]
I've implemented all of Udai's suggestions, and sorted out the rebase thanks to Jaafar. Never using Github Desktop again.

**Review by JaafarRammal — APPROVED** [review]
LGTM thanks Victor.
Please wait for an approval from a team lead who is not on infra team before merging. Thanks!

**Review by MattScottEEE — APPROVED** [review]
This all seems good :)

File: cmd/backend/main.go
Comment (VFjr) [reply cmd/backend/main.go]: I've implemented this now, thanks.

File: config.json
Comment (VFjr) [reply config.json]: I've implemented this now, thanks.

**Review by udai-arneja — APPROVED** [review]
Thanks for the changes

**Pull request merged**
